src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,111 @@
+    private static final int NEGATIVE_AGE = -1;
+    private static final int LARGE_AGE = 234534255;
+    private static final Integer NULL_AGE = null;
+    private static final String VALID_LOGIN = "Antony";
+    private static final String ANOTHER_VALID_LOGIN = "Vlad";
+    private static final String NOT_VALID_NAME = "";
+    private static final String VALID_PASSWORD = "654321";
+    private static final String NOT_VALID_PASSWORD = "111";
+    private RegistrationService registrationService = new RegistrationServiceImpl();
+    private User user = new User();
+
+    @Test
+    void register_defaultUser_NotOk() {
+        assertThrows(InvalidDataException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_LoginNull_NotOk() {
+        user.setLogin(null);
+        user.setPassword(VALID_PASSWORD);
+        user.setAge(VALID_AGE);
+        assertThrows(InvalidDataException.class,() -> registrationService.register(user));
+    }
+
+    @Test
+    void register_ageLessThanEighteen_NotOk() {
+        user.setLogin(VALID_LOGIN);
+        user.setPassword(VALID_PASSWORD);
+        user.setAge(NOT_VALID_AGE);
+        assertThrows(InvalidDataException.class,() -> registrationService.register(user));
+    }
+
+    @Test
+    void register_ageNull_NotOk() {
+        user.setLogin(VALID_LOGIN);
+        user.setPassword(VALID_PASSWORD);
+        user.setAge(NULL_AGE);
+        assertThrows(InvalidDataException.class,() -> registrationService.register(user));
+    }
+
+    @Test
+    void register_passwordNull_NotOk() {
+        user.setLogin(VALID_LOGIN);
+        user.setPassword(null);
+        user.setAge(VALID_AGE);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_notValidPassword_NotOk() {
+        user.setLogin(VALID_LOGIN);
+        user.setPassword(NOT_VALID_PASSWORD);
+        user.setAge(VALID_AGE);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_emptyLogin_NotOk() {
+        user.setLogin(NOT_VALID_NAME);
+        user.setPassword(null);
+        user.setAge(VALID_AGE);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_sameUserInStorage_NotOk() {
+        user.setLogin(VALID_LOGIN);
+        user.setPassword(VALID_PASSWORD);
+        user.setAge(VALID_AGE);
+        registrationService.register(user);
+    }
+
+    @Test
+    void register_nullArgument_NotOk() {
+        assertThrows(InvalidDataException.class, () -> registrationService.register(null));
+    }
+
+    @Test
+    void register_negativeAge_NotOk() {
+        user.setLogin(VALID_LOGIN);
+        user.setPassword(VALID_PASSWORD);
+        user.setAge(NEGATIVE_AGE);
+        assertThrows(InvalidDataException.class,() -> registrationService.register(user));
+    }
```

Comment:
not sure that this check is necessary, because there are tests that will take this option